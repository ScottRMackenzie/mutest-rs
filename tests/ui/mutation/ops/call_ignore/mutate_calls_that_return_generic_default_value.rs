//@ print-mutants
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: call_delete, call_value_default_shadow

#![allow(unused)]

fn f() {
    fn custom_default<T: Default>(_: ()) -> T {
        fn custom_default_impl<T: Default>(_: ()) -> T {
            T::default()
        }

        custom_default_impl::<T>(())
    }

    let default_usize = custom_default::<usize>(());
}

#[test]
fn test() {
    f();
}
