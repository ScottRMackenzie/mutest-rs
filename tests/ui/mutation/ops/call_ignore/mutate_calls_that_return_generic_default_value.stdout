[call_delete] delete call to `f::custom_default` and replace it with `Default::default()` in f at tests/ui/mutation/ops/call_ignore/mutate_calls_that_return_generic_default_value.rs:18:25: 18:52
  <-(0)- test

[call_value_default_shadow] ignore return value of call to `f::custom_default` by shadowing it with `Default::default()` in f at tests/ui/mutation/ops/call_ignore/mutate_calls_that_return_generic_default_value.rs:18:25: 18:52
  <-(0)- test

[call_delete] delete call to `f::custom_default::custom_default_impl` and replace it with `Default::default()` in f::custom_default at tests/ui/mutation/ops/call_ignore/mutate_calls_that_return_generic_default_value.rs:15:9: 15:37
  <-(1)- test

[call_value_default_shadow] ignore return value of call to `f::custom_default::custom_default_impl` by shadowing it with `Default::default()` in f::custom_default at tests/ui/mutation/ops/call_ignore/mutate_calls_that_return_generic_default_value.rs:15:9: 15:37
  <-(1)- test

4 mutations; 4 safe; 0 unsafe (0 tainted)
